subsys/net/ip/net_if.c

```diff
@@ -2529,6 +2531,8 @@ struct net_if_addr *net_if_ipv4_addr_add(struct net_if *iface,
 
 		net_mgmt_event_notify(NET_EVENT_IPV4_ADDR_ADD, iface);
 
+		net_conn_mgr_connect(iface, AF_INET);
+
```

Comment:
So a NET_EVENT_IPV4_ADDR_ADD event is always followed by a net_conn_mgr_connect() and NET_EVENT_IF_DOWN is always followed by a net_conn_mgr_disconnect(). Is there going to be a semantic difference between these two events later on? Right now these NET_EVENT_s can substitute the connect and disconnect events.

Reply:
Currently for IPv4, the ADDR_ADD and IF_DOWN map to CONNECTED/DISCONNECTED events. For IPv6, the situation is more complex as the DAD and/or router added could happen. The idea with CONNECTED/DISCONNECTED is that the application does not need to listen more lower level IPv4/IPv6 specific events, but can listen to more generic events instead.

Reply:
Sounds fine.